parser/ddsmtparser.py

```diff
@@ -1201,36 +1173,17 @@ def is_substvar (self, node):
         return self.scopes.is_substvar(node)
 
     def subst (self, node, substitution):
-        if isinstance (node, SMTScopeNode):
-            self.subst_scopes.subst(node, substitution)
-        elif isinstance (node, SMTCmdNode):
-            self.subst_cmds.subst(node, substitution)
-        else:
-            assert (isinstance (node, SMTNode))
-            self.subst_nodes.subst(node, substitution)
+        self.substs.subst(node, substitution)
+        if isinstance (node, SMTNode):
             if node.is_fun() and self.is_substvar(node):
                 del (self.scopes.declfun_cmds[node.name])
 
 
     def is_subst (self, node):
-        if isinstance (node, SMTScopeNode):
-            return self.subst_scopes.is_subst(node)
-        elif isinstance (node, SMTCmdNode):
-            return self.subst_cmds.is_subst(node)
-        assert (isinstance (node, SMTNode))
-        return self.subst_nodes.is_subst(node)
+        return self.substs.is_subst(node)
 
     def get_subst (self, node):
-        if isinstance (node, SMTScopeNode):
-            assert (not self.subst_scopes.is_subst(node) or
-                        self.subst_scopes.get_subst(node) == None)
-            return self.subst_scopes.get_subst(node)
-        elif isinstance (node, SMTCmdNode):
-            assert (not self.subst_cmds.is_subst(node) or
-                    self.subst_cmds.get_subst(node) == None)
-            return self.subst_cmds.get_subst(node)
-        assert (isinstance (node, SMTNode))
```

Comment:
Can you add the assertions for scope and command nodes back in? They are still valid.